experimental/BasisLieHighestWeight/src/MainAlgorithm.jl

```diff
@@ -462,25 +510,37 @@ function add_by_hand(
   zero_coordinates = compute_zero_coordinates(birational_seq, highest_weight(V))
 
   # calculate new monomials
+  new_monomials = Set{ZZMPolyRingElem}()
   for weight_w in weights_with_non_full_weightspace
     dim_weightspace = weightspaces[weight_w]
-    add_new_monomials!(
-      V,
-      birational_seq,
-      ZZx,
-      matrices_of_operators,
-      monomial_ordering,
-      weightspaces,
-      dim_weightspace,
-      weight_w,
-      monomials_in_weightspace,
-      space,
-      v0,
-      basis,
-      zero_coordinates,
+    union!(
+      new_monomials,
+      add_new_monomials!(
+        V,
+        birational_seq,
+        ZZx,
+        matrices_of_operators,
+        monomial_ordering,
+        weightspaces,
+        dim_weightspace,
+        weight_w,
+        monomials_in_weightspace,
+        space,
+        v0,
+        basis,
+        zero_coordinates,
+      ),
     )
   end
-  return basis
+  return basis, new_monomials
+end
+
+function reduce_to_vertices(possible_vertices::Vector{PointVector{QQFieldElem}})
+  return collect(vertices(convex_hull(possible_vertices)))
```

Comment:
I would guess that most of the additional time is spent in this repeated `convex_hull` and `collect` calls.
Could you try to instead of keeping a list of vertices to keep the polytope itself? This would also simplify some operations, as you can use the Minkowski sum directly